src/Persistence/Array_.php

```diff
@@ -267,128 +277,75 @@ public function export(Model $m, array $fields = null, $typecast_data = true): a
      *
      * @return \atk4\data\Action\Iterator
      */
-    public function initAction(Model $m, $fields = null)
+    public function initAction(Model $model, $fields = null)
     {
-        $keys = $fields ? array_flip($fields) : null;
+        $data = $this->data[$model->table];
 
-        $data = array_map(function ($r) use ($m, $keys) {
-            // typecasting moved to export() method
-            //return $this->typecastLoadRow($m, $keys ? array_intersect_key($r, $keys) : $r);
-            return $keys ? array_intersect_key($r, $keys) : $r;
-        }, $this->data[$m->table]);
+        if ($keys = array_flip((array) $fields)) {
```

Comment:
very, very weird condition to test if `$fields` is not null (or empty, but I belive that is a mistake - we should test simply `!== null`)

Reply:
well we need flipped array, this just does all the work in one line